proposals/lip-0058.md

```diff
@@ -694,29 +765,28 @@ setBFTParameters(precommitThreshold, certificateThreshold, validators):
     bftVotes.activeValidatorsVoteInfo = newActiveValidatorsVoteInfo
 ```
 
-### Endpoints for Off-Chain Services
-
-TBD
 
 ### Genesis Block Processing
 
 The following two steps are executed as part of the genesis block processing, see [LIP 0060][lip-0060] for details.
 
 #### Genesis State Initialization
 
-As part of the genesis state initialization of a genesis block `b`, the BFT module store is initialized as follows:
+As part of the genesis state initialization of a genesis block `b`, the BFT store is initialized as follows:
 
 * The BFT Parameters substore is empty.
 * The BFT Votes substore is initialized as follows:
-  * `bftVotes.maxHeightPrevoted = b.height`,
-  * `bftVotes.maxHeightPrecommitted = b.height`,
-  * `bftVotes.maxHeightCertified = b.height`,
+  * `bftVotes.maxHeightPrevoted = b.header.height`,
+  * `bftVotes.maxHeightPrecommitted = b.header.height`,
+  * `bftVotes.maxHeightCertified = b.header.height`,
   * `bftVotes.blockBFTInfos` is an empty array,
   * `bftVotes.activeValidatorsVoteInfo` is an empty array.
+* The Genesis Data substore is initialized by setting `genesisData.timestamp = h.header.timestamp`.
 
 #### Genesis State Finalization
 
-In this step the BFT module does not execute any logic. However, the exposed function `setBFTParameters` is expected to be called by the PoA module or DPoS module as part of the genesis state finalization of these modules.
+In this step, the function [setValidatorParams][lip-0044#setValidatorParams] of the Validators module should be called by the PoA module or DPoS module.
+These parameters are forwarded to the replication layer and the BFT parameters store is updated accordingly using the function `setBFTParameters` defined above.
 
 ### Block Processing
```

Comment:
I think this section should be removed. As this is not a module anymore, the state machine hooks would not be called. If the logic is still processed, we should clarify when that happens (probably in the replication layer).

Reply:
As concluded in our discussions, I now have a section "Before Application Processing" and "After Application Processing" for the genesis block processing.